Fix(auth): Preserve username in Postgres SSL connections

julienLharidon opened the pull request:
This commit modifies `QgsAuthManager::updateDataSourceUriItems` to retain the original username if it was present in the initial connection string and the authentication method is not one that is expected to provide its own username (e.g., 'basic' auth). This ensures that the user-provided username is always respected.

Changes limited to PG db and non basic connection

Issue confirmed on a pg GCP cloud server I set up
Issue fixed with this PR on my db
Relevant tests : 
- test_core_authmanager passed 
- test_code_datasource_uri passed 
Many other tests failed on my vm. I am still configuring my env. Hope Github tests won't show some regression.

Fixes #60610

nyalldawson commented:
Which auth provider are you wanting to use here? I think this should instead be handled in individual providers' updateDataSourceUriItems implementations, and (where appropriate) exposed as an optional setting for the auth method so that users who don't want this behavior aren't impacted.

Commit:
Fix(auth): Preserve username for identcert

Commit:
Fix(auth): Preserve username for pkipaths

Commit:
Fix(auth): Preserve username for pkcs12

julienLharidon committed:
Fix code formatting with clang-format

julienLharidon commented:
In compliance with QEP 408 AI Tool Policy (still in progress), I add the following tags : Assisted-by:Gemini and Assisted-by: Copilot. 
I chose a small, targeted fix.  I was also able to reproduce the issue by replicating the affected environment. The issue has been successfully reproduced and fixed.
I split the commits accordingly:
- The first commit specifically addresses the reported issue.
- Other auth methods appear to be similar, but I was not able to test them directly.
